sql/builtin_functions.cc

```diff
@@ -1313,6 +1313,59 @@ BoundFunction temporalAggregatorT(const std::vector<BoundSqlExpression> & args)
             GET_ALL};
 }
 
+
+BoundFunction jaccard_index(const std::vector<BoundSqlExpression> & args)
+{
+    if (args.size() != 2)
+        throw HttpReturnException(500, "jaccard_index function takes two arguments");
+
+    return {[=] (const std::vector<ExpressionValue> & args,
+                 const SqlRowScope & scope) -> ExpressionValue
+            {
+                if(!args[0].isRow() || !args[1].isRow())
+                    throw ML::Exception("The arguments passed to the jaccard_index must be two "
+                        "row expressions");
+
+                set<Path> a, b;
+                auto onAtom = [&] (const Path & columnName,
+                                   const Path & prefix,
+                                   const CellValue & val,
+                                   Date atomTs)
+                    {
+                        if (val.empty())
+                            return true;
+
+                        a.insert(columnName);
+                        return true;
+                    };
+
+                args.at(0).forEachAtom(onAtom);
+                b = std::move(a);
+                args.at(1).forEachAtom(onAtom);
+
+                if(a.size() == 0 && b.size() == 0)
+                    return ExpressionValue(1, Date::now());
+
+                vector<Path> intersect(a.size() + b.size());
+                auto it=std::set_intersection(a.begin(), a.end(), b.begin(), b.end(), intersect.begin());
+                intersect.resize(it-intersect.begin());
```

Comment:
Since you dont need the values in intersect anymore, you could simply keep the count. 
ssize_t count = it-intersect.begin();

and use that below